Improved logging

tino1b2be opened the pull request:
Removed the old logging system "twisted.python.log" and replaced it with the new twisted logging system which uses "twisted.logger.Logger".
Added more logging messages for debugging and general logs.

tino1b2be committed:
Added new logs to service.py using twisted.logger
Removed old logs which used twisted.python.log

tino1b2be committed:
Added new logs to utils.py using twisted.logger
Removed old logs which used twisted.python.log
Added a log entry for a xylem_request

tino1b2be committed:
Merge remote-tracking branch 'refs/remotes/origin/fix-mount-path' into improve-logging

tino1b2be committed:
Merge remote-tracking branch 'refs/remotes/origin/fix-mount-path' into improve-logging

tino1b2be committed:
Merge remote-tracking branch 'refs/remotes/origin/fix-mount-path' into improve-logging

# Conflicts:
#	docker_xylem/service.py

tino1b2be committed:
fixed PEP-8 errors

tino1b2be committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into improve-logging

tino1b2be commented:
I've added quite a number of log entries in service.py but not so many in util.py (Don't really know what events to log). Are there any other places/events that need to be logged?

jerith commented on docker_xylem/service.py:
```diff
@@ -36,6 +37,7 @@ def __init__(self, config):
         self.current = {}
 
     def xylem_request(self, queue, call, data):
+        self.log.info('HTTP request to xylem.')
```
Maybe we should add more detail here? The volume name seems like a good thing to know.

tino1b2be replied:
Cool I didn't realise this function was only used when creating a volume.

jerith commented on docker_xylem/service.py:
```diff
@@ -79,16 +81,22 @@ def _umount_fs(self, path):
+        self.log.debug(
+            'Function Call: mount_volume(), `data` = \"{data}\"',
+            data=str(data)
+        )
```
Should we maybe log this in the request dispatcher rather than each method?

tino1b2be replied:
Yes that's a better idea. Will do that.

jerith commented on docker_xylem/service.py:
```diff
@@ -140,25 +174,38 @@ def get_volume_path(self, request, data):
         })
 
         if not result['result']['running']:
+            self.log.error('Error creating volume {name}', name=name)
```
It would be nice to have more detail here. Do we get anything useful from xylem?

tino1b2be replied:
Xylem only returns the name of the volume and the ID. There is no information on any errors.

jerith commented on docker_xylem/service.py:
```diff
@@ -213,14 +272,18 @@ def _route_request(self, request):
+                '{request.path} is not implemented by plugin.',
+                request=request
+            )
             return "Not Implemented"
```
We probably want to return HTTP 404 or something here. Does the docker volume driver documentation say anything about what docker expects?

(Note: This isn't about the logging. If we do decide to change the behaviour, let's open a separate ticket for that.)

tino1b2be replied:
Docker's docs don't specify what response it expects if a plugin does not implement a certain feature.

tino1b2be committed:
Added more detail to the logs.
Removed the logs from the separate functions an added one log in the request dispatcher.

jerith reviewed (approved):
:+1:
